doc/md/language-manual.md

```diff
@@ -1654,6 +1654,24 @@ Such an object literal, sometimes called a *record*, is equivalent to the object
 
 Object expressions support *punning* for concision. A punned field `<id>` is shorthand for `<id> = <id>`; Similarly, a typed, punned field `<id> : <typ>` is short-hand for `<id> = <id> : <typ>`. Both associate the field named `<id>` with the value of the identifier `<id>`.
 
+#### Object reuse and extension
+
+Objects can be combined and/or extended using the `and` and `with` keywords.
+
+A record expression `{ <exp_post> (and <exp_post>)* with <exp-field>;* }` merges the objects (or modules) specified as *bases* which can be
+postfixed expressions, and augments the result to also contain the specified fields. The field list can be omitted when at least two bases appear and none have common field labels.
+Thus the field list serves as
+- disambiguation for field labels occurring more than once in the bases,
+- supplying fresh fields,
+- changing field types, and
+- breaking up entanglement of `var` fields.
+
+The resulting type is determined by the bases' (and explicitly given fields') static type.
+
+`var` fields from the bases (when not disambiguated) will show up in the result as `var` fields that *share* the original contents,
+i.e. an assignment to such a field of either the base or the result will change the shared content. If such a behaviour is not desired,
+then the field must be explicitly overwritten in the explicit field list.
+
```

Comment:
Does that mean I cannot alias a var field from one base when the same name appears in another base?

Reply:
No, you can always _upcast_ the base with the undesired field.

Reply:
Ah, ok. A bit unwieldy, I guess, as you have to write out the full type. But maybe rare enough to be ok in practice.

Reply:
Yes, that is the spirit! :-)

Reply:
You can avoid repeating type signatures by
``` Motoko
type Smaller = { nicefield : Nat, prettyfield : Char };
type Bigger = Smaller and { uglyfield : Text };
let big : Bigger = ...;
// then cast to Smaller:
{ (base : Smaller) and otherbase }
```